### ISSUE TITLE
zlib built with EasyBuild yields "no version information available" error message when it's being used

### ISSUE THREAD

**Issue by boegel** [hpcugent/easybuild-framework#108]
For example:

```
/home/kehoste/.local/easybuild/software/zlib/1.2.7-goalf-1.1.0-no-OFED/lib/libz.so.1: no version information available
```

also see http://superuser.com/questions/305055/how-to-diagnosis-and-resolve-usr-lib64-libz-so-1-no-version-information-avail

This is caused by programs/libraries that both use zlib and other (system) libraries at the same time.

The linker `ld` notices that another library was linked against another version of zlib, and spit out this error message.

Since the error message is printed every time libz.so is loaded, this is annoying.
Also, sometimes it breaks things, for example during the PETSc build (it thinks the `ar` archiver doesn't work properly because the "no version information available" error message is being printed when `ar` is tested).

**Comment by xguse** [comment]
OK I am adding some further information on possible zlib problems.  According to [this thread](https://groups.google.com/forum/#!msg/stacks-users/DJuEmm9e08k/Elm4RotNCQoJ) on the [Stacks](http://creskolab.uoregon.edu/stacks/) user mail list, the base system's zlib seems to be causing confusion during the ./configure process?

relevant post:

>gzbuffer was added to zlib in version 1.2.4. Your compile complains 
about the missing gzbuffer function, but not, say gzopen, which is 
called prior to gzbuffer. This would lead me to believe that you 
potentially have a second zlib.h header on your system that is being 
found during compile prior to the more recent version. However, the 
configure script (which runs before the compile) seems to be finding 
that a newer version is on the system, enabling the use of the gzbuffer 
function, which otherwise would not be included in the compile. 

Specifically, I ended up fixing my problem by listing a specific `zlib` > 1.2.3 in `easyconfig.builddependencies`

```
...
builddependencies = [('zlib', '1.2.7'), ('SAMtools', '1.1'), ('google-sparsehash', '2.0.2')]
...
```

Hope this saves someone else some time or helps inform the discussion.

Gus

**Comment by ebgregory** [comment]
We have had this zlib issue here at JSC as well, mainly stemming from the fact that compilers and linkers are picking up a (very old version of a) system-installed zlib when they are built.

This "no version information available" error seems to show up when a user tries to compile or link a program with a compiler/linker that uses the old system version, while one of the dependencies is the newer EasyBuild-installed newer version

We are dealing with it by moving zlib in the dummy/dummy toolchain as a Core package and making it a (hidden) dependency of binutils, and therefore of the compilers and linkers that are build on top of it.

-Eric

________________________________

**Comment by wpoely86** [comment]
Potential fix is in hpcugent/easybuild-easyconfigs#1350

It would be great if somebody could test it.

**Comment by boegel** [comment]
fixed by https://github.com/hpcugent/easybuild-easyconfigs/pull/1350
